department-of-veterans-affairs/caseflow
# TYLERB/APPEALS-13115: Creation of membership request form on VHA help page

Resolves [APPEALS-13115](https://vajira.max.gov/browse/APPEALS-13115)

### Description
This PR creates a form and related components that allows a user to submit a Membership Request to an organization (vha for now).  It also creates the Vha Help page and relating routing and actions to support that page.

### Acceptance Criteria
- [x] Verify that the newly created help page text and layout matches up with the designs
- [x] General Access Checkbox
  - [x] VHA
- [x] Specialized Access Checkboxes 
  - [x] VHA CAMO
  - [x] VHA Caregiver Support Program
  - [x] Feature Toggle Options
    - [x] Payment Operations Management
    - [x] Veteran and Family Members Program
    - [x] Member Services – Health Eligibility Center
    - [x] Member Services – Beneficiary Travel
    - [x] Prosthetics
- [x] Information Banner if the user is already a member of an organization 
- [x] Optional Text field
- [x] Additional note if the user checks a specialized access checkbox
- [x] Submit button

### Testing Plan
1. Setup data
```ruby
#create a new user
new_user = FactoryBot.create(:user, css_id: "NEW USER")
new_user.save

#Create a user that is a member of the VHA organization
vha_user =FactoryBot.create(:user, css_id: "VHA USER")
vha_user.save
vha_org = BusinessLine.find_by(url: 'vha')
vha_org.add_user(vha_user)
vha_org.save
vha_user.save

#Create a user with pending/active requests
request_user = FactoryBot.create(:user, css_id: "REQUESTS USER")
request_user.save

membership_request = FactoryBot.create(:membership_request, requestor: request_user, organization: VhaCamo.singleton)
membership_request.save

membership_request = FactoryBot.create(:membership_request, requestor: request_user, organization: vha_org)
membership_request.save

membership_request = FactoryBot.create(:membership_request, requestor: request_user, organization: VhaProgramOffice.find_by(name: 'Prosthetics'))
membership_request.save


```
2. Login as the newly created user
3. Navigate to the help page at /help
4. Verify that the new link to the VHA Help is present
![image](https://user-images.githubusercontent.com/109369527/217328447-ea87c1fd-e70b-488b-834c-36de4bb3f406.png)

5. Click the new link to navigate to the VHA Help page 
6. Verify that the page content matches the designs
7. Verify the form checkboxes with the feature toggle disabled. Reload the page after altering the feature toggle
```ruby
FeatureToggle.disable!(:program_office_team_management)
```
![image](https://user-images.githubusercontent.com/109369527/217329248-a93fdbdd-5a10-4be9-b43f-8262979dc00e.png)

8. Verify the form checkboxes are correct with the feature toggle enabled. Reload the page after altering the feature toggle
```ruby
FeatureToggle.enable!(:program_office_team_management)
```
![image](https://user-images.githubusercontent.com/109369527/217330075-a6ec87f8-ddbd-4aa4-bb77-fca459a16714.png)

9.  Verify that the Submit button is disabled if nothing is checked
10. Check the VHA Checkbox and verify that the submit button is enabled
11. Uncheck the box and verify that the submit button is disabled again
12. Check the VHA Caregiver Support checkbox and verify that the submit button is enabled.
13. Additionally, verify that the notice of automatic VHA access appears when a specialized access checkbox is checked. 
![image](https://user-images.githubusercontent.com/109369527/217330075-a6ec87f8-ddbd-4aa4-bb77-fca459a16714.png)
14. Verify that adding text to the optional reason for access text box does not affect the submit button
15. Log in as the created VHA User
16. Navigate back to the Vha help page at /vha/help
17. Verify that the vha checkbox is disabled
![image](https://user-images.githubusercontent.com/109369527/217334572-c9df063d-c2b3-4b9c-90d7-1f9f259ae7dd.png)

18. Also verify that the information alert is displayed indicating that some checkboxes are disabled 
19. Login as CAMOUSER, navigate to the help page, and verify that the camo checkbox is checked
![image](https://user-images.githubusercontent.com/109369527/217331805-36baeb34-0d34-4f0c-8a1a-eac6e75281b7.png)
20. Also verify that the information alert is displayed indicating that some checkboxes are disabled
21. Log in as the created user that has pending/active membership requests and navigate back to the help page
22. Verify that the checkboxes for the Prosthetics, VHA, and VHA CAMO are all disabled.
![image](https://user-images.githubusercontent.com/109369527/217335981-1083447c-39ed-4b9b-8d85-ec889957ec3c.png)
23. Also verify that the information alert is displayed indicating that some checkboxes are disabled



### Storybook Story
* [ ] Added VhaMembershipRequestForm.stories.js
